public/app/core/components/NavBar/NavBarNext.tsx

```diff
@@ -132,6 +132,13 @@ export const NavBarNext: FC = React.memo(() => {
       </NavBarSection>
 
       {showSwitcherModal && <OrgSwitcher onDismiss={toggleSwitcherModal} />}
+      {menuOpen && (
+        <NavBarMenu
+          activeItem={activeItem}
+          navItems={[searchItem, ...coreItems, ...pluginItems, ...configItems]}
```

Comment:
Might be a bit too "premature optimisation", but I think it would be neat to use `useMemo` to spread these, to keep the array identity stable for child components.

Reply:
ok just tried to implement this and it's not trivial. the problem is `configItems` needs the current pathname to generate a sign in link, so we have a dependency on the `location` object, which then causes it to recalculate this pretty much as often as it was before. 🤔